distributed/tests/test_core.py

```diff
@@ -994,3 +996,69 @@ async def long_handler(comm, delay=10):
         await asyncio.wait_for(fut, 0.5)
     await comm.close()
     await server.close()
+
+
+def test_expects_comm():
+    class A:
+        def empty(self):
+            ...
+
+        def one_arg(self, arg):
+            ...
+
+        def comm_arg(self, comm):
+            ...
+
+        def stream_arg(self, stream):
+            ...
+
+        def two_arg(self, arg, other):
+            ...
+
+        def comm_arg_other(self, comm, other):
+            ...
+
+        def stream_arg_other(self, stream, other):
+            ...
+
+        def arg_kwarg(self, arg, other=None):
+            ...
+
+        def comm_posarg_only(self, comm, /, other):
+            ...
+
+        def comm_kwarg_only(self, *, comm, other):
+            ...
+
+        def comm_not_leading_position(self, *, other, comm):
+            ...
+
+        def stream_not_leading_position(self, *, other, stream):
+            ...
+
+        stream_not_leading_partialmethod = partialmethod(
+            stream_not_leading_position, other="foo"
+        )
```

Comment:
This doesn't work because comm/stream was, and is, positional. The whole support for partialmethods is unnecessary.
```
>>> A().stream_not_leading_partialmethod(stream)
TypeError: stream_not_leading_position() takes 1 positional argument but 2 positional arguments (and 1 keyword-only argument) were given
```